src/services/dashboards/widgets/monthly-cost/MonthlyCostWidget.vue

```diff
@@ -3,18 +3,264 @@
+const {
+    createXYDateChart, createXYColumnSeries,
+    createTooltip, createDataProcessor, setChartColors, setXYSingleTooltipText,
+    disposeRoot, refreshRoot,
+} = useAmcharts5(chartContext);
+
+const DATE_FORMAT = 'MMM YYYY';
+const DATE_FIELD_NAME = 'date';
+
 const props = defineProps<WidgetProps>();
-const state = useWidgetState(props);
+
+const state = reactive({
+    ...toRefs(useWidgetState(props)),
+    chartData: computed(() => {
+        if (!state.data) return [];
+        return state.data;
+    }),
+    currentMonthlyCost: 0,
+    previousMonthlyCost: 0,
+    differenceCost: computed<number>(() => state.currentMonthlyCost - state.previousMonthlyCost),
+    differenceCostRate: computed<number>(() => (state.differenceCost !== 0 ? state.differenceCost / state.currentMonthlyCost * 100 : 0)),
+    isDecrease: computed<boolean>(() => (state.differenceCost < 0)),
+    previousMonth: computed<Dayjs>(() => (dayjs.utc(state.options.date_range?.start).subtract(1, 'month'))),
+});
+
+// TODO: api binding
+const fetchData = async () => new Promise((resolve) => {
+    setTimeout(() => {
+        resolve([{
+            date: '2022-01',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-02',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-03',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-04',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-05',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-06',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-07',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-08',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-09',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-10',
+            usd_cost: random(1000, 5000),
+        }, {
+            date: '2022-11',
+            usd_cost: random(1000, 5000),
+        }]);
+    }, 2000);
+});
+
+/* Util */
+const drawChart = (chartData: XYChartData[]) => {
+    const { chart, xAxis, yAxis } = createXYDateChart();
+    setChartColors(chart, state.colorSet);
+    xAxis.get('baseInterval').timeUnit = 'month';
+    const yRendered = yAxis.get('renderer');
+    yRendered.grid.template.setAll({ strokeOpacity: 0 });
+    yRendered.labels.template.setAll({ visible: false });
+
+    chart.setAll({
+        paddingTop: 0,
+        paddingRight: 0,
+        paddingBottom: 0,
+        paddingLeft: -10,
+    });
+
+    const seriesSettings = {
+        name: 'usd_cost',
+        valueYField: 'usd_cost',
+        valueXField: 'date',
+    };
+    const series = createXYColumnSeries(chart, seriesSettings);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    series.columns.template.adapters.add('fill', (fill, target) => {
```

Comment:
I get the error `TS2339: Property 'columns' does not exist on type 'XYSeries'.`, but I don't know the cause. I did ts-igonre once.